inductiva/resources/machines_base.py

```diff
@@ -140,7 +140,24 @@ def n_vcpus(self):
         max_vcpus = int(self.quota_usage["max_vcpus"])
         max_instances = int(self.quota_usage["max_instances"])
 
-        return VCPUCount(max_vcpus, max_vcpus // max_instances)
+        if self.threads_per_core == 2:
```

Comment:
Isn't this the opposite?
Can you maybe write it down in two steps?
1. Compute num vcpus per machine
2. Compute the number of cores per machine 

Also, is the docstring consistent with this new logic?

Reply:
I agree that this is a bit confusing.

I have to do this because the max_vcpus that we get from the back end does not take into consideration the threads per core.

I've tried to simplify things a bit.

Reply:
But then the fix needs to happen in the backend, not here... right?

Reply:
I'm not sure about that. We are getting the values from the quotas. Quotas are related to costs and the cost of a machine is not changed by threads per core. @luispcunha is this correct? This is the reason the max_vcpus is not affected by threads per core right?

Reply:
That would be even more "worrying" -- as this is a property of a VM or physical machine, independently of things like costs or quotas.
We need clear methods that return the number of physical cores and the number of vcpus, ideally for each machine in a machine group. We can start by having that in the MachineGroup (assuming that the machines inside a MachineGroup are "uniform" -- i.e. they are all the same) -- but it ideally it should even be at the level of the machine, as potentially, with the task-runner being launched manually, a machine group can (already) be heterogeneous...

Reply:
But we should do this refactor in another PR no? I want to fix the swan feature asap so that the user can start using it.

Reply:
Ok, then move forward, and open an issue please (if there isn't one yet) for that refactoring (with label technical debt). Thanks!